src/test/ui/consts/operator_traits.rs

```diff
@@ -32,3 +32,31 @@ pub fn foo() {
 pub const fn bar() {
     let x = 42_i32 + 43_i32;
 }
+
+
+#[lang = "Try"]
+#[const_trait]
+pub trait Try: ~const FromResidual<Self::Residual> {
+    type Output;
+    type Residual;
+
+    #[lang = "from_output"]
+    fn from_output(output: Self::Output) -> Self;
+
+    #[lang = "branch"]
+    fn branch(self) -> ControlFlow<Self::Residual, Self::Output>;
+}
+
+#[const_trait]
+// FIXME(keyword_generics): R = <Self as Try>::Residual does not work as it uses `Try<!const>`
+pub trait FromResidual<R> {
+    #[lang = "from_residual"]
+    fn from_residual(residual: R) -> Self;
+}
```

Comment:
I think in order to actually do this we need to support `<X as ~const Trait>` syntax?

I guess the workaround is good enough for libcore for now, most types actually mention the generic parameter on `FromResidual` anyway

Reply:
Should we add a test for the usage of these traits? Especially since the PR on making `Iterator` a `const_trait` was stalled..